src/deadline/client/cli/_groups/manifest_group.py

```diff
@@ -0,0 +1,719 @@
+from urllib.parse import urlparse
+
+import boto3
+import click
+from botocore.client import BaseClient
+
+from deadline.client import api
+from deadline.job_attachments._aws.aws_clients import (
+    get_s3_client,
+    get_s3_transfer_manager,
+)
+from deadline.job_attachments._glob import _process_glob_inputs
+from deadline.job_attachments._utils import _glob_paths
+from deadline.job_attachments.asset_manifests.base_manifest import (
+    BaseAssetManifest,
+    BaseManifestPath,
+)
+from deadline.job_attachments.asset_manifests.decode import decode_manifest
+from deadline.job_attachments.caches import HashCache
+from deadline.job_attachments.download import download_file_with_s3_key
+from deadline.job_attachments.models import (
+    S3_MANIFEST_FOLDER_NAME,
+    AssetRootManifest,
+    GlobConfig,
+    JobAttachmentS3Settings,
+    ManifestDiff,
+    ManifestDownload,
+    ManifestDownloadResponse,
+)
+from deadline.job_attachments.upload import FileStatus, S3AssetManager, S3AssetUploader
+
+from ...config import config_file
+from ...exceptions import NonValidInputError
+from .._common import _apply_cli_options_to_config, _handle_error, _ProgressBarCallbackManager
+from .click_logger import ClickLogger
+
+
+@click.group(name="manifest")
+@_handle_error
+def cli_manifest():
+    """
+    Commands to work with AWS Deadline Cloud Job Attachments.
+    """
+
+
+@cli_manifest.command(name="snapshot")
+@click.option("--root", required=True, help="The root directory to snapshot. ")
+@click.option(
+    "--destination",
+    default=None,
+    help="Destination directory where manifest is created. Defaults to the root directory.",
+)
+@click.option(
+    "--name",
+    default=None,
+    help="Name of the manifest. A time stamp is added YYYY-MM-DD-HH-MM-SS for versioning.",
+)
+@click.option(
+    "--glob",
+    default=None,
+    help="Glob include and exclude of directory and file regex to include in the manifest.",
+)
+@click.option("--diff", default=None, help="Asset Manifest to diff against.")
+@click.option("--json", default=None, is_flag=True, help="Output is printed as JSON for scripting")
+@_handle_error
+def manifest_snapshot(
+    root: str, destination: str, name: str, glob: str, diff: str, json: bool, **args
+):
+    """
+    Creates manifest of files specified by root directory.
+    """
+    logger: ClickLogger = ClickLogger(is_json=json)
+    if not os.path.isdir(root):
+        raise NonValidInputError(f"Specified root directory {root} does not exist. ")
+
+    if destination and not os.path.isdir(destination):
+        raise NonValidInputError(f"Specified destination directory {destination} does not exist. ")
+    elif destination is None:
+        destination = root
+        logger.echo(f"Manifest creation path defaulted to {root} \n")
+
+    glob_config: GlobConfig = _process_glob_inputs(glob)
+    inputs = _glob_paths(root, include=glob_config.include_glob, exclude=glob_config.exclude_glob)
+
+    # Placeholder Asset Manager
```

Comment:
What does the placeholder here means? Or we need an empty constructor for `S3AssetManager`?